src/inmanta/parser/plyInmantaParser.py

```diff
@@ -736,49 +736,74 @@ def p_constant_mls(p: YaccProduction) -> None:
     if len(matches) == 0:
-        return Literal(str(string))
-
-    offset = 3 if mls else 1  # len(""")  or len(') or len(")
-    init: List[str]
-    match: List[str]
-    locatable_matches: List(LocatableString) = []
-    for obj in matches:
-        init, match, _ = (str(string)[start:end].splitlines() for start, end in pairwise((0, obj[1], obj[2], len(obj))))
-        start_line: int = string.lnr + len(init) - 1
-        end_line: int = start_line + len(match) - 1
-        len_last_init_line: int = len(init[-1]) if init else 0
-        start_char: int = len(init[-1]) + offset + string.start if len(init) == 1 else len_last_init_line + 1
-        end_char: int = len(match[-1]) + start_char - 1 if len(match) == 1 else len(match[-1])
-        range: Range = Range(string.location.file, start_line, start_char, end_line, end_char)
-        locatable_string = LocatableString("".join(match), range, string.lexpos, string.namespace)
-        locatable_matches.append(locatable_string)
-    return create_string_format(string, locatable_matches)
-
-
-def create_string_format(format_string: LocatableString, variables: List[LocatableString]) -> StringFormat:
+        return Literal(str(string_ast))
+
+    mls_offset: int = 3 if mls else 1  # len(""")  or len(') or len(")
+    locatable_matches: List[Tuple[str, LocatableString]] = []
+    for match in matches:
+        init_string: str = str(string_ast)[0 : match.start() + 2]
+        match_string: str = match[2]
+        init_lines: List[str] = (init_string + "\n").splitlines()  # lines before the match
+        match_lines: List[str] = match_string.splitlines()  # lines matching the string format
+        line_offset, char_offset = get_offset(match_lines, match[3])
+        line: int = string_ast.lnr + len(init_lines) + line_offset - 1
+        start_char: int = (
+            char_offset
+            if len(match_lines) > 1
+            else string_ast.start + len(init_lines[-1]) + mls_offset + char_offset - 1
+            if len(init_lines) == 1
+            else len(init_lines[-1]) + char_offset
+        )
+        end_char: int = start_char + len(match[3])
+        range: Range = Range(string_ast.location.file, line, start_char, line, end_char)
+        locatable_string = LocatableString(match[3], range, string_ast.lexpos, string_ast.namespace)
+        locatable_matches.append((match[1], locatable_string))
+    return create_string_format(string_ast, locatable_matches)
+
+
+def get_offset(match_lines: List[str], name: str) -> Tuple[int, int]:
+    for line_index in range(len(match_lines)):
+        match: Optional[re.Match[str]] = re.search(name, match_lines[line_index])
+        if match:
+            return line_index, match.start() + 1
+    else:
+        raise Exception("Match not found")
+
+
+def create_string_format(format_string: LocatableString, variables: List[Tuple[str, LocatableString]]) -> StringFormat:
```

Comment:
If I read this correctly, it assumes that the 'variables' never contain a newline. 

Do we have a testcase to verify this is effectively not allowed?

Reply:
It is indeed assumed. I just tested it I don't think so. even worst it is accepted by the compiler 
```
a ="""{{this.is
.the.string}}"""
std::print(a)
```
will print the string
 ```
"{{this.is
.the.string}}"
```
I'll add a new ticket when this is merged

Reply:
I think your example is flawed. The regex does not allow newlines so it is not recognized as a string format, hence it prints the literal string. I do think it's a good idea to add this invariant to the docstring though.